public/app/features/alerting/unified/components/rule-editor/AnnotationKeyInput.tsx

```diff
@@ -19,7 +19,7 @@ interface Props {
 }
 
 export const AnnotationKeyInput: FC<Props> = ({ value, onChange, existingKeys, width, className }) => {
-  const [isCustom, setIsCustom] = useState(false);
+  const [isCustom, setIsCustom] = useState(!!value && !Object.keys(AnnotationOptions).includes(value));
```

Comment:
nit: maybe break out this statement in a helper function that is a bit easier to understand at glance.

Reply:
- [ ] break it up